code/noRAM/app/src/main/java/com/example/noram/AttendeeProfileFragment.java

```diff
@@ -87,37 +89,101 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
-        email.setText(attendee.getEmail());
-        allowLocation.setChecked(attendee.getAllowLocation());
+        setFields(attendee, view);
 
         // Save the entered information when the save button is clicked
         view.findViewById(R.id.attendee_info_save_button).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                attendee.setFirstName(firstName.getText().toString());
-                attendee.setLastName(lastName.getText().toString());
-                attendee.setHomePage(homePage.getText().toString());
-                attendee.setEmail(email.getText().toString());
-                attendee.setAllowLocation(allowLocation.isChecked());
+                String editFirstName = firstName.getText().toString();
+                String editLastName = lastName.getText().toString();
+                String editHomePage = homePage.getText().toString();
+                String editEmail = email.getText().toString();
+                Boolean editAllowLocation = allowLocation.isChecked();
+
+                // Validate name and email fields
+                if (validateAttendeeFields(editFirstName, editLastName, editEmail)) {
+                    attendee.setFirstName(editFirstName);
+                    attendee.setLastName(editLastName);
+                    attendee.setHomePage(editHomePage);
+                    attendee.setEmail(editEmail);
+                    attendee.setAllowLocation(editAllowLocation);
+                }
             }
         });
 
         // Revert the changes when the cancel button is clicked
         view.findViewById(R.id.attendee_info_cancel_button).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                firstName.setText(attendee.getFirstName());
-                lastName.setText(attendee.getLastName());
-                homePage.setText(attendee.getHomePage());
-                email.setText(attendee.getEmail());
-                allowLocation.setChecked(attendee.getAllowLocation());
+                setFields(attendee, view);
             }
         });
-
         return view;
     }
+
+    /**
+     * This method is called when the fragment is resumed.
+     * It sets the fields to the attendee's information,
+     * ensuring that any unsaved changes are reset.
+     */
+    @Override
+    public void onResume() {
+        super.onResume();
+        View view = getView();
+        attendee = MainActivity.attendee;
+
+        if (view != null) {
+            setFields(attendee, view);
+        }
+    }
+
+    /**
+     * Set the fields of the view to the attendee's information
+     * @param attendee the attendee whose information is being displayed
+     * @param view the view that is being displayed
+     */
+    public void setFields(Attendee attendee, View view) {
+        // Get the necessary fields from the view
```

Comment:
nit: could make these properties of the fragment so we don't have to call findViewById every time we want to set the fields